build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/BwcSetupExtension.java

```diff
@@ -137,10 +158,13 @@ private TaskProvider<LoggedExec> createRunBwcGradleTask(
         });
     }
 
-    private String minimumCompilerVersionPath(Version bwcVersion) {
-        return (bwcVersion.onOrAfter(BUILD_TOOL_MINIMUM_VERSION))
-            ? "build-tools-internal/" + MINIMUM_COMPILER_VERSION_PATH
-            : "buildSrc/" + MINIMUM_COMPILER_VERSION_PATH;
+    /** A convenience method for getting java home for a version of java and requiring that version for the given task to execute */
+    private static String getJavaHome(ObjectFactory objectFactory, JavaToolchainService toolChainService, final int version) {
```

Comment:
I think we should return a `Provider` here. It otherwise becomes too easy to call these methods and accidentally resolve toolchains early.

Reply:
done